libraries/AP_Scripting/examples/rover-TerrainDetector.lua

```diff
@@ -0,0 +1,192 @@
+--[[----------------------------------------------------------------------------
+
+TerrainDetector ArduPilot Lua script
+
+Uses gyro and accelerometer data to detect rough terrain and slow the
+vehicle's speed accordingly during auto missions.
+
+CAUTION: This script is capable of engaging and disengaging autonomous control
+of a vehicle.  Use this script AT YOUR OWN RISK.
+
+-- Yuri -- Jun 2022
+
+LICENSE - GNU GPLv3 https://www.gnu.org/licenses/gpl-3.0.en.html
+
+Concept first presented during this live stream:
+https://www.youtube.com/watch?v=UdXGXjigxAo&t=7155s
+Credit to @ktrussell for the idea and discussion!
+------------------------------------------------------------------------------]]
+
+local SCRIPT_NAME     = 'TerrainDetector'
+local RUN_INTERVAL_MS =  25  -- needs to be pretty fast for good detection
+local SBY_INTERVAL_MS = 500  -- slower interval when detection is disabled
+
+-- tuning values
+local SCR_USER_SPEED_PARAM     = 'SCR_USER1'  -- (m/s) set rough terrain speed here

```

Comment:
Would be good to load a custom param name. It would be good to be able to remove the scirpting user params at some point.

Reply:
@IamPete1, custom param set added per discussion.